src/async_graphql_hyper.rs

```diff
@@ -128,6 +128,52 @@ impl GraphQLResponse {
         Ok(response)
     }
 
+    fn default_body(&self) -> Result<Body> {
+        Ok(Body::from(serde_json::to_string(&self.0)?))
+    }
+
+    pub fn to_response(self) -> Result<Response<hyper::Body>> {
+        self.build_response(StatusCode::OK, self.default_body()?)
+    }
+
+    fn flatten_response(data: Value) -> Value {
+        if let Value::Object(map) = data.clone() {
+            if map.len() == 1 {
+                if let Some(value) = map.values().next() {
+                    return value.clone();
+                }
+            }
+        }
+        data
```

Comment:
That's unreachable. I'm keeping it here as fallback, but we should never reach this point.